### PR TITLE
Developer guide updates after API modules changes

### PR DESCRIPTION
**Pull request by arkadius**

### CODE REVIEW

File: docs/developers_guide/Basics.md [hunk 1]
```diff
@@ -28,24 +26,32 @@ There are following types of components:
 - `SinkFactory`
 - `CustomStreamTransformer` - types of transformations depend on type of Engine
 - `Service` - mainly for defining stateless enrichments
+To read more see [ComponentProvider API](./Components)
 
+## Execution of scenarios
+
+Designer use [DeploymentManager](https://github.com/TouK/nussknacker/blob/staging/ui/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/DeploymentManager.scala)
```
Comment (mproch) [inline docs/developers_guide/Basics.md:33]:
```suggestion
The Designer uses [DeploymentManager](https://github.com/TouK/nussknacker/blob/staging/ui/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/DeploymentManager.scala)
```

File: docs/developers_guide/Basics.md [hunk 2]
```diff
@@ -28,24 +26,32 @@ There are following types of components:
...
+interface to perform actions on scenarios (deploy / cancel / etc.). All available in distribution deployment providers are located in `managers` directory and are added to designer classpath.
```
Comment (mproch) [inline docs/developers_guide/Basics.md:34]:
```suggestion
interface to perform actions on scenarios (deploy / cancel / etc.). All providers that are available in distribution deployment are located in `managers` directory and are added to the designer classpath.
```

File: docs/developers_guide/Basics.md [hunk 3]
```diff
@@ -28,24 +26,32 @@ There are following types of components:
 - `SinkFactory`
 - `CustomStreamTransformer` - types of transformations depend on type of Engine
 - `Service` - mainly for defining stateless enrichments
+To read more see [ComponentProvider API](./Components)
 
+## Execution of scenarios
```
Comment (mproch) [inline docs/developers_guide/Basics.md:31]: Execution or deployment?

File: docs/developers_guide/Basics.md [hunk 4]
```diff
@@ -54,32 +60,46 @@ There are following types of components:
...
   - [OAuth2ServiceFactory](https://github.com/TouK/nussknacker/blob/staging/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2ServiceFactory.scala)
 - [CountsReporterCreator](https://github.com/TouK/nussknacker/blob/staging/ui/processReports/src/main/scala/pl/touk/nussknacker/processCounts/CountsReporter.scala)
+- [NodeAdditionalInfoProvider](https://github.com/TouK/nussknacker/blob/staging/extensions-api/src/main/scala/pl/touk/nussknacker/engine/additionalInfo/NodeAdditionalInfoProvider.scala)
+- [CustomProcessValidatorFactory](https://github.com/TouK/nussknacker/blob/staging/ui/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/validation/CustomProcessValidator.scala)
+             
+## Modules architecture and conventions
 
-                         
-## Contents of customization packages.
+Diagram below shows dependencies between modules. You can see on it two main groups:
```
Comment (mproch) [inline docs/developers_guide/Basics.md:68]:
```suggestion
The diagram below shows dependencies between modules. You can see two main groups on it :
```

File: docs/developers_guide/Basics.md [hunk 5]
```diff
@@ -54,32 +60,46 @@ There are following types of components:
...
+- API modules
+- Utils
 
-Customization code should be placed in jar files on correct classpath:
-- Customizations of model (in particular `ComponentProviders`) should be configured in [Model config](../installation_configuration_guide/Configuration) in 
-`modelConfig.classPath`. 
-- Code of Designer customizations should go to the main designer classpath (e.g. put the jars in the `lib` folder)
+*API modules* contains interfaces that are needed by our core modules (on both designer and runtime side).
+
+In the other hand *Utils* modules contains classes build on top of API which can be used in extensions but are not mandatory. **API of *Utils* modules can be changed more often then API inside API modules**
```
Comment (mproch) [inline docs/developers_guide/Basics.md:74]:
```suggestion
On the other hand *Utils* modules contain classes built on top of API which can be used in extensions but are not mandatory. **API of *Utils* modules can be changed more often than API inside API modules**
```

File: docs/developers_guide/Basics.md (hunk 3 above)
Comment (arkadius) [reply docs/developers_guide/Basics.md]: I've replaced with "Deployment of scenarios"

**Review by mproch — APPROVED** [review]

**Pull request merged**